src/golang/lib/workflow/validate.go

```diff
@@ -0,0 +1,77 @@
+package workflow
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/shared"
+	"github.com/aqueducthq/aqueduct/lib/collections/workflow"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/repos"
+	"github.com/aqueducthq/aqueduct/lib/workflow/utils"
+	"github.com/dropbox/godropbox/errors"
+)
+
+const (
+	internalValidationErrMsg = "Internal system error occurred while validating Workflow schedule."
+)
+
+// ValidateSchedule validates the provided schedule. The following conditions are
+// not allowed:
+// - Having a CascadingUpdateTrigger where SourceID is for a Workflow that is
+// running on a non self-orchestrated engine, such as Airflow. This is not allowed
+// since Aqueduct cannot trigger Workflow runs at the end of execution on an
+// engine that is not self-orchestrated.
+// It returns an HTTP status code and a client-friendly error, if any.
+func ValidateSchedule(
+	ctx context.Context,
+	schedule workflow.Schedule,
+	engineType shared.EngineType,
+	artifactRepo repos.Artifact,
+	dagRepo repos.DAG,
+	dagEdgeRepo repos.DAGEdge,
+	operatorRepo repos.Operator,
+	workflowRepo repos.Workflow,
+	DB database.Database,
+) (int, error) {
+	if schedule.Trigger != workflow.CascadingUpdateTrigger {
+		// Only CascadingUpdateTriggers require validation
+		return http.StatusOK, nil
+	}
+
+	if engineType == shared.AirflowEngineType {
```

Comment:
@cw75 This is the error checking we are discussing on #861

Reply:
thx!